src/server/src/main/java/io/cassandrareaper/jmx/JmxProxyImpl.java

```diff
@@ -848,30 +890,19 @@ public boolean isConnectionAlive() {
       String connectionId = getConnectionId();
       return null != connectionId && connectionId.length() > 0;
     } catch (IOException e) {
-      LOG.error("Couldn't get Connection Id", e);
+      LOG.debug("Couldn't get Connection Id", e);
+      return false;
     }
-    return false;
   }
 
-  /**
-   * Cleanly shut down by un-registering the listener and closing the JMX connection.
-   */
+  public void removeRepairStatusHandler(int commandId) {
+    repairStatusHandlers.remove(commandId);
+  }
+
+  /** Cleanly shut down by un-registering the listener and closing the JMX connection. */
   @Override
   public void close() {
-    LOG.debug("close JMX connection to '{}': {}", host, jmxUrl);
-    if (this.repairStatusHandler.isPresent()) {
-      try {
-        mbeanServer.removeNotificationListener(ssMbeanName, this);
-        LOG.debug("Successfully removed notification listener for '{}': {}", host, jmxUrl);
-      } catch (InstanceNotFoundException | ListenerNotFoundException | IOException e) {
-        LOG.debug("failed on removing notification listener", e);
-      }
-    }
-    try {
-      jmxConnector.close();
-    } catch (IOException e) {
-      LOG.warn("failed closing a JMX connection", e);
-    }
+    // Not closing anymore, we re-use JMX connections now
```

Comment:
this seems odd. this presents an API that is no longer accurate.
the right thing to here is to remove the `close()` method, and from the `AutoClosable` interface from `JmxProxy`.
that involves a handful more lines of code change (moving the resource declaration in try-resource statements onto the subsequent line inside the try-catch block).

Reply:
I've removed the extension of AutoClosable in JmxProxy and fixed all try-catch-resource blocks involving it.
`close()` is back and only used in the `JmxConnectionFactory` to close defunct JMX connections.